File: src/JavaScript/access_token.js
"use strict"

const application_id = document.getElementById("application_id");
const application_secret = document.getElementById("application_secret");
const button_token = document.getElementById("button_token");
const access_token = document.getElementById("access_token");

button_token.onclick = async () => {
    try {
        access_token.value = await getAccessToken(application_id.value, application_secret.value);
    } catch (error) {
        mylog(error);
    }
}

async function getAccessToken(application_id, application_secret) {

    const AUTH_SERVER_URI = "https://auth.mimi.fd.ai/v2/token";
    const SCOPE = "https://apis.mimi.fd.ai/auth/asr/http-api-service;https://apis.mimi.fd.ai/auth/asr/websocket-api-service;https://apis.mimi.fd.ai/auth/nict-asr/http-api-service;https://apis.mimi.fd.ai/auth/nict-asr/websocket-api-service;https://apis.mimi.fd.ai/auth/nict-tts/http-api-service;https://apis.mimi.fd.ai/auth/nict-tra/http-api-service";

    const response = await fetch(AUTH_SERVER_URI, {
        method: "POST",
        body: new URLSearchParams({
            "client_id": application_id,
            "client_secret": application_secret,
            "grant_type": "https://auth.mimi.fd.ai/grant_type/application_credentials",
            "scope": SCOPE
        })
    })

    if (!response.ok) {
        throw Error("failed to issue an access token");
    }

    const data = await response.json();
    const token = data["accessToken"];

    return token;
}

File: src/Python/access_token.py
import requests
import sys
import logging

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

AUTH_SERVER_URI = 'https://auth.mimi.fd.ai/v2/token'
APPLICATION_ID = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'  # Fix me
APPLICATION_SECRET = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'  # Fix me
SCOPE = 'https://apis.mimi.fd.ai/auth/asr/http-api-service;https://apis.mimi.fd.ai/auth/asr/websocket-api-service;https://apis.mimi.fd.ai/auth/nict-asr/http-api-service;https://apis.mimi.fd.ai/auth/nict-asr/websocket-api-service;https://apis.mimi.fd.ai/auth/nict-tts/http-api-service;https://apis.mimi.fd.ai/auth/nict-tra/http-api-service'


def get_access_token():
    resp, status = http_request()
    if status != 200:
        logger.error('failed to get an access token')
        sys.exit(1)
    access_token = resp['accessToken']
    return access_token


def http_request():
    data = {
        'client_id': APPLICATION_ID,
        'client_secret': APPLICATION_SECRET,
        'grant_type': 'https://auth.mimi.fd.ai/grant_type/application_credentials',
        'scope': SCOPE
    }
    url = AUTH_SERVER_URI
    resp = requests.post(url, data=data)
    return (resp.json(), resp.status_code)


if __name__ == '__main__':
    logger.info('getting an access token...')
    print(get_access_token())
